web/src/components/feature/ComponentTable.tsx

```diff
@@ -0,0 +1,83 @@
+import { ComponentResponse } from '../../api/generated'
+import { Autocomplete } from '@equinor/eds-core-react'
+import { useState } from 'react'
+import {
+  DynamicEditableTable,
+  TDynamicEditableTableInput,
+} from '../common/DynamicEditableTable'
+import styled from 'styled-components'
+
+type componentEntry = {
+  componentId: string
+  altName: string
+  chemicalFormula: string
+}
+
+const ComponentSelectorContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  row-gap: 4px;
+  max-width: 500px;
+`
+
+const ComponentTableContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 400px;
+  overflow: auto;
+`
+
+function transformRawResponse(
+  selectedComponents: Array<componentEntry>
+): TDynamicEditableTableInput {
+  const tableInput: TDynamicEditableTableInput = {
+    [`Component`]: { editable: false, values: [] },
+    [`Feed value (mol)`]: { editable: true, values: [] },
+  }
+  // Push data to each phase column
+  selectedComponents.forEach(({ componentId, altName, chemicalFormula }) => {
+    tableInput[`Component`].values.push(`${altName} (${chemicalFormula})`)
+    tableInput[`Feed value (mol)`].values.push()
+  })
+  return tableInput
+}
+
+export const ComponentSelector = ({
+  components,
+}: {
+  components: ComponentResponse
+}) => {
+  const [selectedEntries, setSelectedEntries] = useState(Array<componentEntry>)
+
+  function createOptions(components: ComponentResponse) {
```

Comment:
Can and should be moved out of component if it's not refering to any "class" vars.